cdap-common/src/main/java/co/cask/cdap/common/http/DefaultHttpRequestConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.http;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.common.http.HttpRequestConfig;
+
+import java.net.HttpURLConnection;
+
+/**
+ * Class to uniformly configure CDAP HTTP requests with a user-configured timeout
+ */
+public class DefaultHttpRequestConfig extends HttpRequestConfig {
+
+  public DefaultHttpRequestConfig() {
+    super(Integer.getInteger(Constants.HTTP_CLIENT_CONNECTION_TIMEOUT_MS),
```

Comment:
Also, if the property is somehow missing (meaning someone is not using the `ConfigModule`), then this will throw a strange NPE. Better check it explicitly (using a private static help method).